libraries/botbuilder-core/botbuilder/core/message_factory.py

```diff
@@ -0,0 +1,166 @@
+        """
+        Returns a message that will display a set of attachments in list form.
+
+        :Example:
+        message = MessageFactory.list([CardFactory.hero_card(HeroCard(title='title1',
+                                                             images=[CardImage(url='imageUrl1')],
+                                                             buttons=[CardAction(title='button1')])),
+                                       CardFactory.hero_card(HeroCard(title='title2',
+                                                             images=[CardImage(url='imageUrl2')],
+                                                             buttons=[CardAction(title='button2')])),
+                                       CardFactory.hero_card(HeroCard(title='title3',
+                                                             images=[CardImage(url='imageUrl3')],
+                                                             buttons=[CardAction(title='button3')]))])
+        await context.send_activity(message)
+
+        :param attachments:
+        :param text:
+        :param speak:
+        :param input_hint:
+        :return:
+        """
+        return attachment_activity(AttachmentLayoutTypes.list, attachments, text, speak, input_hint)
+
+    @staticmethod
+    def carousel(attachments: List[Attachment], text: str = None, speak: str = None,
+                 input_hint: Union[InputHints, str] = None) -> Activity:
+        """
+        Returns a message that will display a set of attachments using a carousel layout.
+
+        :Example:
+        message = MessageFactory.carousel([CardFactory.hero_card(HeroCard(title='title1',
+                                                                 images=[CardImage(url='imageUrl1')],
+                                                                 buttons=[CardAction(title='button1')])),
+                                           CardFactory.hero_card(HeroCard(title='title2',
+                                                                 images=[CardImage(url='imageUrl2')],
+                                                                 buttons=[CardAction(title='button2')])),
+                                           CardFactory.hero_card(HeroCard(title='title3',
+                                                                 images=[CardImage(url='imageUrl3')],
+                                                                 buttons=[CardAction(title='button3')]))])
+        await context.send_activity(message)
+
+        :param attachments:
+        :param text:
+        :param speak:
+        :param input_hint:
+        :return:
+        """
+        return attachment_activity(AttachmentLayoutTypes.carousel, attachments, text, speak, input_hint)
+
+    @staticmethod
+    def content_url(url: str, content_type: str, name: str = None, text: str = None, speak: str = None,
+                    input_hint: Union[InputHints, str] = None):
```

Comment:
so we are still keeping this one overload?
- I'm thinking of simplifying the C# to just take the object in cases like this.
- but we can check in with this and resolve that later

Reply:
Actually we need to keep that to account for the _not-Swagger-defined_ `"delay"`-type activity. Without this overload we'd be saying no to this specific activity type.